fe/fe-core/src/main/java/com/starrocks/mysql/privilege/Auth.java

```diff
@@ -827,6 +827,29 @@ public void replayGrant(PrivInfo privInfo) {
     private void grantInternal(UserIdentity userIdent, String role, TablePattern tblPattern,
                                PrivBitSet privs, boolean errOnNonExist, boolean isReplay)
             throws DdlException {
+
+        if (! isReplay) {
+            // check privilege only on leader, in case of replaying old journal
+            switch (tblPattern.getPrivLevel()) {
+                case DATABASE:
+                    if (privs.containsNodePriv() || privs.containsResourcePriv() || privs.containsImpersonatePriv()) {
+                        throw new DdlException(
+                                "Db privilege can not contains global or resource or impersonate privileges: " + privs);
+                    }
+                    break;
+
+                case TABLE:
+                    if (privs.containsNodePriv() || privs.containsResourcePriv() || privs.containsImpersonatePriv()) {
+                        throw new DdlException(
```

Comment:
```suggestion
                        throw new DdlException("Table does not contains {}", privs);
```
The `Table privilege can not contains global or resource or impersonate privileges` is misunderstanding.